webviews/components/history_view.svelte

```diff
@@ -1,16 +1,175 @@
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_CANCELED:
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_TIMER_STARTED:
+          // TODO: append as a command
+          wfts[wfts.length - 1].events.push(ev)
+          break
+        default:
+          currWFT.events.push(ev)
+          break
+      }
+    }
+    console.log(wfts)
+    return wfts
+  }
+
+  //get labelTextForHistoryEvent
+  function labelTextForHistoryEvent(history: temporal.api.history.v1.IHistoryEvent) {
+    const { eventType } = history
+    if (eventType === undefined || eventType === null) {
+      throw new TypeError("Expected history event `eventType` to be defined")
+    }
+    const eventTypeName: string = temporal.api.enums.v1.EventType[eventType]
+      .replace(/^EVENT_TYPE_/, "")
+      .split("_")
+      .map((p) => `${p[0]}${p.substr(1).toLowerCase()}`)
+      .join("")
+    currentEvent = eventType
+    console.log(currentEvent)
+    switch (eventType) {
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED:
+        startTime = tsToDate(history.eventTime).getTime() / 1000
+        return ` ⇨ ${eventTypeName} (${history.workflowExecutionStartedEventAttributes?.workflowType?.name})`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        console.log(tsToDate(history.eventTime).getTime() / 1000, "-", startTime, "=", duration)
+        return ""
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_FAILED:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        // console.log((tsToDate(history.eventTime).getTime()/1000),"-",startTime,"=",duration)
+        return `⇨ ${eventTypeName}`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        // console.log((tsToDate(history.eventTime).getTime()/1000),"-",startTime,"=",duration)
+        return `⇨ ${eventTypeName}`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_CONTINUED_AS_NEW:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        // console.log((tsToDate(history.eventTime).getTime()/1000),"-",startTime,"=",duration)
+        return `⇨ ${eventTypeName}`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_SCHEDULED:
+        return ` ⇦ ${eventTypeName} (${history.activityTaskScheduledEventAttributes?.activityType?.name})`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_STARTED:
+        if (history.activityTaskStartedEventAttributes?.lastFailure) {
+          return ` ⇦ ${eventTypeName} (${history.activityTaskStartedEventAttributes?.lastFailure.message})`
```

Comment:
The message could possibly be too long to display here, put a TODO comment to address this later please